weblate/vcs/git.py

```diff
@@ -544,57 +544,32 @@ def push(self):
         self.execute(['svn', 'dcommit', self.branch])
 
 
-class GithubRepository(GitRepository):
+class _GitRemoteRepositoryMixin(GitRepository):
 
-    name = 'GitHub'
-
-    _cmd = 'hub'
+    # Subclasses must override _username.
+    _username = None
 
     _is_supported = None
     _version = None
 
     @classmethod
     def is_supported(cls):
-        if not settings.GITHUB_USERNAME:
+        if not cls._username:
             return False
-        return super(GithubRepository, cls).is_supported()
-
-    @classmethod
-    def _getenv(cls):
-        """Generate environment for process execution."""
-        env = super(GithubRepository, cls)._getenv()
-        # Add path to config if it exists
-        userconfig = os.path.expanduser('~/.config/hub')
-        if os.path.exists(userconfig):
-            env['HUB_CONFIG'] = userconfig
-
-        return env
-
-    def create_pull_request(self, origin_branch, fork_branch):
-        """Create pull request to merge branch in forked repository into
-        branch of remote repository.
-        """
-        cmd = [
-            'pull-request',
-            '-f',
-            '-h', '{0}:{1}'.format(settings.GITHUB_USERNAME, fork_branch),
-            '-b', origin_branch,
-            '-m', settings.DEFAULT_PULL_MESSAGE,
-        ]
-        self.execute(cmd)
+        return super(_GitRemoteRepositoryMixin, cls).is_supported()
 
     def push_to_fork(self, local_branch, fork_branch):
         """Push given local branch to branch in forked repository."""
-        if settings.GITHUB_USERNAME:
-            cmd_push = ['push', '--force', settings.GITHUB_USERNAME]
+        if self._username:
```

Comment:
I know this code was originally there, but can you please remove this condition? It's not useful anymore (I'm too lazy to dig out original use case in git history). The class will not be used without username being available.

Reply:
Sure, will do!

Reply:
I've done this in 3d7cc2ddf2, you will have to rebase, sorry.

Reply:
Done.